quality_of_service_demo/rclcpp/src/message_lost_listener.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
+
+#include "sensor_msgs/msg/image.hpp"
+
+#include "quality_of_service_demo/visibility_control.h"
+
+namespace quality_of_service_demo
+{
+class MessageLostListener : public rclcpp::Node
+{
+public:
+  QUALITY_OF_SERVICE_DEMO_PUBLIC
```

Comment:
It's needed because the executable created by `rclcpp_components` macro is dynamically loading the node at runtime.
Thus the constructor must be a public symbol.